app/src/main/java/com/internship/move/presentation/menu/MenuFragment.kt

```diff
@@ -18,26 +19,35 @@ class MenuFragment : Fragment(R.layout.fragment_menu) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        initObserver()
-
-        binding.clearApp.setOnClickListener {
-            viewModel.clearApp()
-            findNavController().navigate(MenuFragmentDirections.actionMenuFragmentToSplashGraph())
+        viewModel.getUserRequest("Bearer ${viewModel.getLoginToken()}")
+        viewModel.userData.observe(viewLifecycleOwner) {
+            binding.toolbar.title = getString(R.string.menu_appbar_heading) + " ${it.name}!"
         }
 
-        binding.backIV.setOnClickListener{
-            findNavController().navigate(MenuFragmentDirections.actionMenuFragmentToMapFragment())
-        }
+        initListeners()
     }
 
-    private fun initObserver() {
-        binding.logoutBtn.setOnClickListener {
-            viewModel.logout(viewModel.getLoginToken())
-            viewModel.loggedOut.observe(viewLifecycleOwner) {
-                if (it) {
-                    findNavController().navigate(MenuFragmentDirections.actionMenuFragmentToSplashGraph())
-                }
-            }
+    private fun initListeners() {
+        binding.toolbar.setNavigationOnClickListener {
```

Comment:
I prefer leaving an empty line between setters. It makes the code a bit clearer imo